src/YoutubeAtom/YoutubeAtom.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+
+import { YoutubeOverlay } from './YoutubeOverlay';
+import { MaintainAspectRatio } from './MaintainAspectRatio';
+import { YoutubeAtomType } from '../types';
+
+type EmbedConfig = {
+    adsConfig: {
+        adTagParameters: {
+            iu: string;
+            cust_params: string;
+        };
+    };
+};
+
+interface AdTargeting {
+    adUnit: string;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    customParams: { [key: string]: any };
+}
+
+const buildEmbedConfig = (adTargeting: AdTargeting): EmbedConfig => {
+    return {
+        adsConfig: {
+            adTagParameters: {
+                iu: `${adTargeting.adUnit || ''}`,
+                cust_params: encodeURIComponent(
+                    constructQuery(adTargeting.customParams),
+                ),
+            },
+        },
+    };
+};
+
+const constructQuery = (query: { [key: string]: any }): string =>
```

Comment:
I suppose there isn't anywhere more sensible for this to live?

Reply:
Yeah, it is pretty generic code. I'm not too concerned about this being duplicated though (if it does end up being useful elsewhere), as it's only a few lines of code. Or else to refactor later as an alternative.